Emby.Server.Implementations/Library/MediaStreamSelector.cs

```diff
@@ -13,39 +11,24 @@ namespace Emby.Server.Implementations.Library
 {
     public static class MediaStreamSelector
     {
-        public static int? GetDefaultAudioStreamIndex(List<MediaStream> streams, string[] preferredLanguages, bool preferDefaultTrack)
+        public static int? GetDefaultAudioStreamIndex(IReadOnlyList<MediaStream> streams, IReadOnlyList<string> preferredLanguages, bool preferDefaultTrack)
         {
-            streams = GetSortedStreams(streams, MediaStreamType.Audio, preferredLanguages)
-               .ToList();
+            var sortedStreams = GetSortedStreams(streams, MediaStreamType.Audio, preferredLanguages);
 
             if (preferDefaultTrack)
             {
-                var defaultStream = streams.FirstOrDefault(i => i.IsDefault);
-
-                if (defaultStream != null)
-                {
-                    return defaultStream.Index;
-                }
-            }
-
-            var stream = streams.FirstOrDefault();
-
-            if (stream != null)
-            {
-                return stream.Index;
+                return sortedStreams.FirstOrDefault(i => i.IsDefault)?.Index;
```

Comment:
Old behavior: if there is no default stream it falls through and returns the first stream
New behavior: if there is no default stream it returns null, even if there is a non-default stream

Was this intended?

Reply:
Nice catch, fixed and added test for this case.